edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/CertificateRenewal.cs

```diff
@@ -33,12 +34,13 @@ public CertificateRenewal(EdgeHubCertificates certificates, ILogger logger)
                 // Clamp the renew time to TimeSpan.FromMilliseconds(Int32.MaxValue)
                 // This is the maximum value for the timer (~24 days)
                 // Math.Min unfortunately doesn't work with TimeSpans so we need to do the check manually
-                TimeSpan withBuffer = timeToExpire - TimeBuffer;
-                TimeSpan renewAfter = withBuffer > MaxRenewAfter
+                TimeSpan renewAfter = timeToExpire - (TimeBuffer / 2);
```

Comment:
The TimeBuffer/2 is so that the Callback goes to the if case and not the else case?

Reply:
It's to make sure that on the final iteration it falls within the buffer window. The buffer is there so that we don't actually ever expire the cert.

But, if we calculate the next timer fire based on the start of the window, we could fall right before it. Using the `TimeBuffer / 2` I think will force to be fully within the window so that it will always hit the else on the last iteration.